cdap-client/src/main/java/co/cask/cdap/client/NamespaceClient.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright © 2014 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.client;
+
+import co.cask.cdap.client.config.ClientConfig;
+import co.cask.cdap.client.exception.AlreadyExistsException;
+import co.cask.cdap.client.exception.BadRequestException;
+import co.cask.cdap.client.exception.CannotBeDeletedException;
+import co.cask.cdap.client.exception.NotFoundException;
+import co.cask.cdap.client.exception.UnAuthorizedAccessTokenException;
+import co.cask.cdap.client.util.RESTClient;
+import co.cask.cdap.proto.NamespaceMeta;
+import co.cask.common.http.HttpMethod;
+import co.cask.common.http.HttpRequest;
+import co.cask.common.http.HttpResponse;
+import co.cask.common.http.ObjectResponse;
+import com.google.common.reflect.TypeToken;
+import com.google.gson.Gson;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.List;
+import javax.inject.Inject;
+
+/**
+ * Client to interact with CDAP namespaces
+ */
+public class NamespaceClient {
+  private final RESTClient restClient;
+  private final ClientConfig config;
+
+  private static final String NAMESPACE_ENTITY_TYPE = "namespace";
+
+  @Inject
+  public NamespaceClient(ClientConfig config) {
+    this.config = config;
+    this.restClient = RESTClient.create(config);
+  }
+
+  /**
+   * Lists all namespaces in CDAP.
```

Comment:
I *could* be wrong about this, but I think it is incorrect to say "namespace in CDAP". I think the convention is "Lists all namespaces in a CDAP instance" or just "Lists all namespaces".